src/util/basic-functions.js

```diff
@@ -0,0 +1,16 @@
+import { getContrastRatio } from '@material-ui/core/styles/colorManipulator'
+
+/**
+ * get white or black color for text based on background color
+ *
+ * @export
+ * @param {string} backgroundColor
+ */
+export function determineTextColor(backgroundColor) {
+  const ratioToWhite = getContrastRatio(backgroundColor, '#FFF')
+  const ratioToBlack = getContrastRatio(backgroundColor, '#000')
+
+  const textColor = ratioToWhite + 2 >= ratioToBlack ? '#FFF' : '#000'
```

Comment:
This is Cool 👍  I have a question about the `textColor` — why does `ratioToWhite` have 2 added?

Reply:
because some color such as the primary one return ratioToBlack 5.2 and ratioToWhite 4.6 and also for Orange so by adding 2 fix bumping it up fix the issue

Reply:
I see. Perhaps adding 2 might be favouring the white colour when it does not meet minimum accessibility standard. Do you think we can set a threshold? For instance, we can conditionally add 2 when `ratioToWhite` is higher than 2.5

Reply:
Also, we can convert a hardcoded value to a variable so that it can be more flexible to future changes. But regardless I like the amount of thoughts you put into it! Didn't think about the case of changing contrast ratio when hover 👍

Reply:
mmm makes sense. will see what I can do there